apps/src/templates/progress/progressHelpers.js

```diff
@@ -194,3 +206,87 @@ export const processedLevel = level => {
       level.sublevels && level.sublevels.map(level => processedLevel(level))
   };
 };
+
+const getLevelResult = serverProgress => {
+  if (serverProgress.status === LevelStatus.locked) {
+    return TestResults.LOCKED_RESULT;
+  }
+  if (serverProgress.readonly_answers) {
+    return TestResults.READONLY_SUBMISSION_RESULT;
+  }
+  if (serverProgress.submitted) {
+    return TestResults.SUBMITTED_RESULT;
+  }
+
+  return serverProgress.result || resultFromStatus(serverProgress.status);
+};
+
+/**
+ * Parse a level progress object that we get from the server using either
+ * /api/user_progress or /dashboardapi/section_level_progress into our
+ * canonical studentLevelProgressType shape.
+ * @param {object} serverObject A progress object from the server
+ * @returns {studentLevelProgressType} Our canonical progress shape
+ */
+export const levelProgressFromServer = serverObject => {
+  return {
+    status: serverObject.status || LevelStatus.not_tried,
+    result: getLevelResult(serverObject),
```

Comment:
Could you elaborate a bit on what status and result mean and how they are used?

Reply:
see the type definition in `progressTypes.studentLevelProgressType` and let me know if the comments there require elaboration.